code/modules/events/grid_check.dm

```diff
@@ -1,11 +1,20 @@
-/datum/event/grid_check	//NOTE: Times are measured in master controller ticks!
-	announceWhen		= 5
+#define ANNOUNCE_DELAYED FALSE
+#define ANNOUNCE_NOW TRUE
+
+#define ANNOUNCE_AFTER_MC_TICKS 5
+#define APC_TOGGLE_OFF_PROBABILITY 25
+#define EVENT_FIXES_ITSELF_AT_END FALSE
+#define EVENT_MAX_LENGTH_MC_TICKS 150
+#define EVENT_MIN_LENGTH_MC_TICKS 60
+
+/datum/event/grid_check
+	announceWhen = ANNOUNCE_AFTER_MC_TICKS
 
 /datum/event/grid_check/setup()
-	endWhen = rand(30,120)
+	endWhen = rand(EVENT_MIN_LENGTH_MC_TICKS, EVENT_MAX_LENGTH_MC_TICKS)
 
 /datum/event/grid_check/start()
-	power_failure(0)
+	power_failure(ANNOUNCE_DELAYED)
```

Comment:
Possibly change this to `power_failure(announce=ANNOUNCE_DELAYED)` for ease of readability.